src/OrleansServiceBus/Providers/Streams/EventHub/SegmentBuilder.cs

```diff
@@ -80,7 +80,7 @@ public static void Append(ArraySegment<byte> segment, ref int writerOffset, stri
             else
             {
                 var bytes = new byte[str.Length * sizeof(char)];
-                Array.Copy(str.ToCharArray(), 0, bytes, 0, bytes.Length);
+                Buffer.BlockCopy(str.ToCharArray(), 0, bytes, 0, bytes.Length);
```

Comment:
I believe Array.Copy copies array elements which doesn't work because one array is characters (uint16) and the other is bytes.  Buffer.BlockCopy copies the raw memory (bytes) so it's fine, but it pins the buffers.  I'd like to avoid using Buffer.BlockCopy, but for now I'm just rolling back to it because I know it works.